db/src/db.rs

```diff
@@ -982,6 +985,48 @@ impl MentatStoring for rusqlite::Connection {
         results.map(|_| ())
     }
 
+    /// Insert datoms corresponding to :db/retractEntity entities into the search results.
+    fn insert_retract_entities(&self, entids: &[Entid]) -> Result<()> {
```

Comment:
I don't think we want to allow retracting:

- Core vocabulary.
- Attributes.
- Transactions.

Could you add validation here?